src/modules/Account/Devices.ts

```diff
@@ -272,6 +279,63 @@ class Devices extends TagoIOModule<GenericModuleParams> {
 
     return result;
   }
+
+  /**
+   * Edit data records in a device using the profile token and device ID.
+   *
+   * The `updatedData` can be a single data record or an array of records to be updated,
+   * each of the records must have the `id` of the record and the fields to be updated.
+   *
+   * @param deviceId Device ID.
+   * @param updatedData A single or an array of updated data records.
+   *
+   * @returns Success message indicating amount of records updated (can be 0).
+   *
+   * @example
+   * ```ts
+   * const myDevice = new Device({ token: "my_device_token" });
+   *
+   * await myDevice.editDeviceData("myDeviceId", { id: "idOfTheRecord", value: "new value", unit: "new unit" });
+   * ```
+   */
+  public async editDeviceData(deviceId: GenericID, updatedData: DataEdit | DataEdit[]): Promise<string> {
+    const result = await this.doRequest<string>({
+      path: `/device/${deviceId}/data`,
+      method: "PUT",
+      body: updatedData,
+    });
+
+    return result;
+  }
+
+  /**
+   * Delete data records in a device using the profile token and device ID.
+   *
+   * See the example to understand how to use this method properly to have full control on what to delete.
+   *
+   * ! If query parameters are empty, all data for the device will be deleted.
```

Comment:
This is an important part of the documentation, we handle this in our usage of the method but it needed to be clear to the user that not sending filters like `id` might result in the entire bucket's data being deleted.